### PR TITLE
Add soft restarts for minimal downtime

### PR DESCRIPTION
**Pull request by Brainicism**

### CODE REVIEW

File: src/scripts/announce-restart.ts [hunk 1]
```diff
@@ -54,6 +71,8 @@ process.on("SIGINT", async () => {
     const restartDate = new Date();
     restartDate.setMinutes(restartDate.getMinutes() + restartMinutes);
 
+    console.log(options);
```
Comment (taahamahdi) [inline src/scripts/announce-restart.ts:74]:
```suggestion
```

File: .github/workflows/deploy-prod.yml [hunk 2]
```diff
@@ -1,45 +1,54 @@
...
-        uses: actions/checkout@v2
+    snap:
+        runs-on: ubuntu-latest
+        timeout-minutes: 10
+        steps:
+            - name: Checkout KMQ_Discord
+              uses: actions/checkout@v2
+              with:
+                  fetch-depth: 0
+                  token: ${{ secrets.PAT }}
```
Comment (taahamahdi) [inline .github/workflows/deploy-prod.yml:30]: I think I missed where secrets.PAT is declared

**Review by taahamahdi — CHANGES_REQUESTED** [review]

File: src/scripts/announce-restart.ts (hunk 1 above)
Comment (Brainicism) [reply src/scripts/announce-restart.ts:74]: It's a nice to have

File: .github/workflows/deploy-prod.yml (hunk 2 above)
Comment (Brainicism) [reply .github/workflows/deploy-prod.yml:30]: This wasn't added in this commit, it's in the repo's secret key vault

**Review by taahamahdi — APPROVED** [review]

**Pull request merged**